server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -512,13 +514,13 @@ public synchronized IndexShard createShard(
                 }
             };
             Directory directory = directoryFactory.newDirectory(this.indexSettings, path);
-            Directory remoteDirectory = null;
             RemoteStoreRefreshListener remoteStoreRefreshListener = null;
             if (this.indexSettings.isRemoteStoreEnabled()) {
                 try {
                     Repository repository = repositoriesService.repository(clusterService.state().metadata().clusterUUID());
-                    remoteDirectory = remoteDirectoryFactory.newDirectory(this.indexSettings, path, repository);
-                    remoteStoreRefreshListener = new RemoteStoreRefreshListener(directory, remoteDirectory);
+                    Directory remoteDirectory = remoteDirectoryFactory.newDirectory(this.indexSettings, path, repository);
+                    ScheduledThreadPoolExecutor executor = new Scheduler.SafeScheduledThreadPoolExecutor(1);
```

Comment:
This logic will not work. It will create one executor per shard. Let me re-think on this.